js/background.js

```diff
@@ -483,40 +502,55 @@
         });
     }
 
-    function onMessageReceived(ev) {
-        var data = ev.data;
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            var profileKey = data.message.profileKey.toArrayBuffer();
-            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
-              return sender.setProfileKey(profileKey).then(ev.confirm);
-            });
-        }
-        var message = initIncomingMessage(data);
-
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
+  /* eslint-enable */
+  /* jshint ignore:start */
+  async function onMessageReceived(ev) {
+    const { data } = ev;
+
+    // eslint-disable-next-line no-bitwise
+    if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
+      const profileKey = data.message.profileKey.toArrayBuffer();
+      const sender = ConversationController.getOrCreateAndWait(data.source, 'private');
+      await sender.setProfileKey(profileKey);
+      // TODO: Is `ev.confirm` a `Promise`? Original code returned it:
+      return ev.confirm();
```

Comment:
Ported the code as verbatim as possible and noticed this discrepancy with `ev.confirm` and down below. Let me know what you think.

Reply:
I suppose the return value of `ev.confirm()` is a promise, so let's return it.

Reply:
Thanks — done!